code-challenges.test.js

```diff
@@ -13,38 +13,214 @@
+//   ✓ takes in an array, removes the first item from the array and shuffles the remaining content (2 ms)
+// netVotes
+//   ✓ takes an object of up votes and down votes and returns the net total of votes (1 ms)
+// noDuplicates
+//   ✕ takes in two arrays as arguments and returns one array with no duplicate values
+
+// ● noDuplicates › takes in two arrays as arguments and returns one array with no duplicate values
+
+//   ReferenceError: noDuplicates is not defined
 
 
 // b) Create the function that makes the test pass.
+
+    // PSEUDOCODE:
+        // declare a function named noDuplicates
+        // takes in two arrays as arguments
+            // combine the two arrays into one
+            // iterate through the combined array
+        // returns one array with no duplicate values
+
+// const noDuplicates = (arrOne, arrTwo) => {
+//     let combinedArr = arrOne.concat(arrTwo)
+//     let finalArr = []
+//     for (let i = 0; i < arrOneTwo.length; i++) {
+//         if (finalArr.includes(arrOneTwo[i]) === false) {
+//             finalArr.push(arrOneTwo[i])
+//         }
+//     } return finalArr
+// }
+
+// PASS  ./code-challenges.test.js
+// removeAndShuffle
+//   ✓ takes in an array, removes the first item from the array and shuffles the remaining content (2 ms)
+// netVotes
+//   ✓ takes an object of up votes and down votes and returns the net total of votes
+// noDuplicates
+//   ✓ takes in two arrays as arguments and returns one array with no duplicate values
+
+// REFACTORED CODE:
+
+// const noDuplicates = (arrOne, arrTwo) => {
+//     let combinedArr = arrOne.concat(arrTwo)
+//     combinedArr = combinedArr.filter((value, index) => {
+//         return combinedArr.indexOf(value) === index
+//     })
+//     return combinedArr
+// }
+
+// PASS  ./code-challenges.test.js
+// removeAndShuffle
+//   ✓ takes in an array, removes the first item from the array and shuffles the remaining content (2 ms)
+// netVotes
+//   ✓ takes an object of up votes and down votes and returns the net total of votes (1 ms)
+// noDuplicates
+//   ✓ takes in two arrays as arguments and returns one array with no duplicate values (1 ms)
+
+//STRETCH:
+
+    // PSEUDOCODE:
+        // declare a function named noDuplicates
+        // takes in any number of arguments using the spread operator (arguments will be placed in an array)
+            // declare a variable that will be the concatenated array of all array arguments
+            // iterate through the arrays to concatenate them
+            // filter the combined array for duplicates
+        // returns a single array with no duplicate values
+
+const noDuplicates = (...multArr) => {
+    for (let i =0; i < multArr.length - 1; i++) {
+        let combinedArr = multArr[i].concat(multArr[i+1])
+        combinedArr = combinedArr.filter((value, index) => {
+            return combinedArr.indexOf(value) === index
+            })
+        return combinedArr
+    }
+    return combinedArr
+}
```

Comment:
This is great. You tried different ways. I would suggest trying to use the spread operator in your logic. You may find that it will help you truly refactor. I also suggest searching "javascript how to remove duplicates from an array". You will see another method that will significantly refactor your function.